server/cpho/util.py

```diff
@@ -22,3 +30,25 @@ def group_by(iterable, key):
 
 
 flatten = lambda l: [item for sublist in l for item in sublist]
+
+
+class classproperty(property):
+    def __get__(self, cls, owner):
+        return classmethod(self.fget).__get__(None, owner)()
+
+
+class GroupFetcher:
+    @classproperty
+    @lru_cache
+    def admin_group(cls):
+        return Group.objects.get_or_create(name=ADMIN_GROUP_NAME)[0]
+
+    @classproperty
+    @lru_cache
+    def account_manager_group(cls):
+        return Group.objects.get_or_create(name=ACCOUNT_MANAGER_GROUP_NAME)[0]
+
+    @classproperty
+    @lru_cache
+    def hso_group(cls):
+        return Group.objects.get_or_create(name=HSO_GROUP_NAME)[0]
```

Comment:
trying something new here, it's a neat shorthand and involves at most 3 queries, it'll even get cached across multiple requests (which is OK because these are totally static)